### DISCUSSION TITLE
Azure Blob Storage sink issues with compression

### DISCUSSION THREAD

**Discussion by danelkotev** [vectordotdev/vector#22074] [Q&A] [answered]
When using Azure Blob Storage sink  with `gzip` compression, there are issues with Azure - when downloading the file to my local machine, it seems like the file is corrupted.
There are many issues with Azure around it from what I see online, however when removing `CONTENT-ENCODING` all seems OK, any change I can override the headers added by vector.dev although keep using  `gzip` compression? any other suggestion will gladly accepted...

**Comment by jszwedko** [comment]
@danelkotev how are you downloading the file? It may be being transparently decompressed by your browser. This gotcha is noted here: https://vector.dev/docs/reference/configuration/sinks/azure_blob/#compression

**Reply by danelko** [reply]
when downloading it from azure browser to my mac it fails to decompress, claiming the file compression method is unknown, but the issue is that even in azure blob storage it has issue

**Reply by jszwedko** [reply, marked as answer]
Are you sure the downloaded file was compressed? The browser will transparently decompress when downloading so you may have ended up with the uncompressed file on your local filesystem. You can use the `file` command locally to verify the type of the file.

**Reply by danelko** [reply]
seems like the browser decompress them and leave the extension as-is.

can we think of a feature that allows us to uploaded compressed files without the headers or without the extension?

**Reply by pront** [reply]
I think you can achieve this by setting https://vector.dev/docs/reference/configuration/sinks/azure_blob/#compression to `none`.

**Reply by danelko** [reply]
But I do want to use compression, I do want to store uncompressed files

**Reply by pront** [reply]
I see. You probably want something like this: https://github.com/vectordotdev/vector/issues/21795#issuecomment-2479140185
